hyperspy/_signals/signal1d.py

```diff
@@ -1157,17 +1181,17 @@ def remove_background(
             elif background_type in ('SkewNormal', 'Skew Normal'):
                 background_estimator = components1d.SkewNormal()
             else:
-                raise ValueError(
-                    "Background type: " +
-                    background_type +
-                    " not recognized")
-            spectra = self._remove_background_cli(
+                raise ValueError(f"Background type: {background_type} not "
+                                 "recognized.")
+            result = self._remove_background_cli(
                 signal_range=signal_range,
                 background_estimator=background_estimator,
                 fast=fast,
                 zero_fill=zero_fill,
-                show_progressbar=show_progressbar)
-            return spectra
+                show_progressbar=show_progressbar,
+                model=model,
+                return_model=return_model)
+            return result
```

Comment:
If I understood correctly, we've now got - if interactive, return model - if not interactive, return (result, model) ?

I can see why it would be hard to do anything else here but I think that at least needs to be documented pretty carefully in the Returns section of the docstring.